src/coreclr/src/jit/emitarm64.cpp

```diff
@@ -14882,4 +14887,35 @@ emitter::insExecutionCharacteristics emitter::getInsExecutionCharacteristics(ins
 
 #endif // defined(DEBUG) || defined(LATE_DISASM)
 
+//----------------------------------------------------------------------------------------
+// IsOppositeOfPrevMov:
+//    Check if previous instruction was a mov where dst was current src and src was current dst.
+//
+// Arguments:
+//    ins - The current instruction
+//    dst - The current destination
+//    src - The current source
+//
+// Return Value:
+//    true if previous instruction moved from current dst to src.
+
+bool emitter::IsOppositeOfPrevMov(instruction ins, regNumber dst, regNumber src)
+{
+    assert(ins == INS_mov);
+
+    if (emitLastIns != nullptr && emitLastIns->idIns() == INS_mov)
```

Comment:
Suggest you only do this if opts are enabled.

Reply:
Sorry, how do I check? Is `OptimizationEnabled()` the right way?

Reply:
Yes, `emitComp->opts.OptimizationEnabled()`

Though I would check it in the caller of this method.